tests/test_annotations.py

```diff
@@ -40,6 +40,32 @@ def setUp(self):
             self.CLUSTER_PATH, ['text/csv', 'text/plain', 'text/tab-separated-values']
         )
 
+    def test_low_mem_artifact(self):
+        # pandas default of low_memory=True allows internal chunking during parsing
+        # causing inconsistent dtype coercion artifact for larger annotation files
+
+        lmtest = Annotations(
+            '../tests/data/low_mem_unit.txt',
+            ['text/csv', 'text/plain', 'text/tab-separated-values'],
+        )
+        lmtest.preprocess()
+
+        # when low memory=True, the first row in the file would be in the first chunk
+        # and the numeric value was not properly coerced to become a string
+        assert isinstance(
+            lmtest.file['mixed_data']['group'][0], str
+        ), "numeric value should be coerced to string"
+        # This test will need to be adjusted when SCP-2545 is implemented so NA values
+        # become strings for group annotations. This test checks for current behavior,
+        # an empty cell becomes a NaN value which is a float.
+        assert isinstance(
+            lmtest.file['mixed_data']['group'][2], float
+        ), "expect empty cell conversion to NaN which is a float"
+        # numeric value in second chunk should still properly be coerced to string type
+        assert isinstance(
+            lmtest.file['mixed_data']['group'][32800], str
```

Comment:
Empty newlines between these blocks would help readability:

```suggestion
            lmtest.file['mixed_data']['group'][0], str
        ), "numeric value should be coerced to string"
        
        # This test will need to be adjusted when SCP-2545 is implemented so NA values
        # become strings for group annotations. This test checks for current behavior,
        # an empty cell becomes a NaN value which is a float.
        assert isinstance(
            lmtest.file['mixed_data']['group'][2], float
        ), "expect empty cell conversion to NaN which is a float"
        
        # numeric value in second chunk should still properly be coerced to string type
        assert isinstance(
            lmtest.file['mixed_data']['group'][32800], str
```

Reply:
Yes, thanks for the suggestion. `9457765`